docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/ops_util.py

```diff
@@ -0,0 +1,195 @@
+        shard_id=shard_id,
+        num_shards=num_shards,
+        random_shuffle=random_shuffle,
+    )
+
+    images = dali.fn.decoders.image(
+        inputs["image/encoded"],
+        device="mixed" if device == "gpu" else "cpu",
+        output_type=dali.types.RGB,
+    )
+    xmin = inputs["image/object/bbox/xmin"]
+    xmax = inputs["image/object/bbox/xmax"]
+    ymin = inputs["image/object/bbox/ymin"]
+    ymax = inputs["image/object/bbox/ymax"]
+    bboxes = dali.fn.transpose(dali.fn.stack(xmin, ymin, xmax, ymax), perm=[1, 0])
+    classes = dali.fn.cast(inputs["image/object/class/label"], dtype=dali.types.INT32)
+    return (
+        images,
+        bboxes,
+        classes,
+        dali.fn.cast(inputs["image/width"], dtype=dali.types.FLOAT),
+        dali.fn.cast(inputs["image/height"], dtype=dali.types.FLOAT),
+    )
+
+
+def input_coco(
+    images_path, annotations_path, device, shard_id, num_shards, random_shuffle=True
+):
+    encoded, bboxes, classes = dali.fn.readers.coco(
+        file_root=images_path,
+        annotations_file=annotations_path,
+        ratio=True,
+        ltrb=True,
+        shard_id=shard_id,
+        num_shards=num_shards,
+        random_shuffle=random_shuffle,
+    )
+
+    images = dali.fn.decoders.image(
+        encoded,
+        device="mixed" if device == "gpu" else "cpu",
+        output_type=dali.types.RGB,
+    )
+
+    shape = dali.fn.peek_image_shape(encoded, type=dali.types.FLOAT)
+    heights = dali.fn.slice(shape, 0, 1, axes=(0,))
+    widths = dali.fn.slice(shape, 1, 1, axes=(0,))
+
+    return (
+        images,
+        bboxes,
+        classes,
+        widths,
+        heights,
+    )
+
+
+def normalize_flip(device, images, bboxes, p=0.5):
+    flip = dali.fn.random.coin_flip(probability=p)
+    images = dali.fn.crop_mirror_normalize(
+        images,
+        mirror=flip,
+        mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+        std=[0.229 * 255, 0.224 * 255, 0.225 * 255],
+        output_layout=dali.types.NHWC,
+        device=device,
+    )
+    bboxes = dali.fn.bb_flip(bboxes, horizontal=flip, ltrb=True)
+    return images, bboxes
+
+
+def gridmask(device, images, widths, heights):
+
+    p = dali.fn.random.coin_flip()
+    ratio = 0.4 * p
+    angle = dali.fn.random.normal(mean=-1, stddev=1) * 10.0 * (math.pi / 180.0)
+
+    l = dali.math.min(0.5 * heights, 0.3 * widths)
+    r = dali.math.max(0.5 * heights, 0.3 * widths)
+    tile = dali.fn.cast(
+        (dali.fn.random.uniform(range=[0.0, 1.0]) * (r - l) + l),
+        dtype=dali.types.INT32,
+    )
+
+    gridmask = dali.fn.grid_mask(
+        images, ratio=ratio, angle=angle, tile=tile, device=device
+    )
+
+    return images
+
+
+def random_crop_resize(
+    device, images, bboxes, classes, widths, heights, output_size, scaling=[0.1, 2.0]
+):
+
+    if scaling is None:
+        scale_factor = 1.0
+    else:
+        scale_factor = dali.fn.random.uniform(range=scaling)
+
+    sizes = dali.fn.cat(widths, heights)
```

Comment:
The size for `resize` is in height, width order
```suggestion
    sizes = dali.fn.cat(heights, widths)
```

Reply:
Done